cabal-install-solver/src/Distribution/Solver/Modular/Message.hs

```diff
@@ -54,38 +116,42 @@ showMessages = go 0
     go :: Int -> Progress Message a b -> Progress String a b
     go !_ (Done x)                           = Done x
     go !_ (Fail x)                           = Fail x
+    -- TODO: I should use the the level of recursion to change the displayMessage' indentation level ...
+
     -- complex patterns
     go !l (Step (TryP qpn i) (Step Enter (Step (Failure c fr) (Step Leave ms)))) =
         goPReject l qpn [i] c fr ms
+
```

Comment:
Do you think it's possible to separate the part of this function that applies the `goPReject` and `goPSkip` special handlers to the part that applies `displayMessage` to all messages? They are separate concerns and that is why we where doing this refactor (IIRC).

Reply:
I guess this can be done another time.